### PR TITLE
feat(bedrock): Implement Prompt management

### LINKED ISSUE #664: (bedrock): Add prompt and prompt management
### Describe the feature

https://docs.aws.amazon.com/bedrock/latest/userguide/prompt-management.html 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-bedrock-prompt.html
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-bedrock-promptversion.html

### Use Case

Missing feature

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by aws-rafams**
Fixes #664

- Implements Prompt Management
- Integration test can be run with command `npx projen integ:prompts:deploy`.
- Due to the addition of integration tests, `@aws-cdk/integ-tests-alpha` was added as a dev dependency in `.projenrc.ts`
- Update projen to trigger snapshot update when deps are upgraded
- Linting rules

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Comment by krokoko** [comment]
Thanks @aws-rafams ! Qq: "Fixes partially" -> is this because of the current limitation mentioned in [this ticket](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2115), so this PR implements Prompt/Prompt management from the original ticket, correct ? If yes, I'll create a separate ticket for Prompt/Prompt management and link it to this PR

**Comment by krokoko** [comment]
Could you please add code snippets for the new feature to https://github.com/aws-rafams/generative-ai-cdk-constructs/blob/prompt-management/src/cdk-lib/bedrock/README.md ?

**Comment by aws-rafams** [comment]
> Thanks @aws-rafams ! Qq: "Fixes partially" -> is this because of the current limitation mentioned in [this ticket](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/2115), so this PR implements Prompt/Prompt management from the original ticket, correct ? If yes, I'll create a separate ticket for Prompt/Prompt management and link it to this PR

Hey Alain!
Correct, this is because prompt management can be implemented regardless of the Prompt Flow part which I'm waiting for the CFN fix for the missing Node Types.

File: .projenrc.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
     'typedoc-plugin-markdown',
     'aws-sdk-mock',
     '@aws-cdk/assert',
+    '@aws-cdk/integ-tests-alpha',
```
Comment (krokoko) [inline .projenrc.ts:67]: should we add the explicit version here, to make sure it matches the cdk version ? CDK_VERSION-alpha.0

Comment (aws-rafams) [reply .projenrc.ts:67]: good catch! Yes, will update it accordingly

Comment (aws-rafams) [reply .projenrc.ts]: just updated the code to include this with the latest commit

**Review by krokoko — DISMISSED** [review]

**Comment by aws-rafams** [comment]
need to change some things, let's pause the merge

**Comment by aws-rafams** [comment]
desired changes done, thanks :)

**Comment by krokoko** [comment]
@aws-rafams ha you were faster, I was pushing right after you to update the snapshots :) thanks !

**Review by krokoko — DISMISSED** [review]

**Comment by scottschreckengaust** [comment]
Thank you for the contribution!

nit: please consider keeping formatting separate whenever possible to ease reviews of form verse function.

File: .gitignore [hunk 2]
```diff
@@ -56,6 +56,15 @@ junit.xml
 !/.eslintrc.json
 .jsii
 tsconfig.json
+test/cdk-lib/bedrock/integ-tests/.tmp
+test/cdk-lib/bedrock/integ-tests/prompts.integ.snapshot/asset.*
```
Comment (scottschreckengaust) [inline .gitignore:60]: I don't see a convention adopted for the outputs during integration testing at https://docs.aws.amazon.com/cdk/api/v2/docs/integ-tests-alpha-readme.html, but which of these may allow a convention to adopt if these are added to other namespaces? I'm also curious where the `manifest.json` and `tree.json` come from?

```
test/cdk-lib/**/.tmp
test/cdk-lib/**/asset.*
test/cdk-lib/**/cdk.out
```

File: package.json [hunk 3]
```diff
@@ -43,12 +49,14 @@
...
   "devDependencies": {
     "@aws-cdk/assert": "^2.68.0",
+    "@aws-cdk/integ-tests-alpha": "2.154.1-alpha.0",
     "@commitlint/config-conventional": "^18.6.3",
     "@mrgrain/jsii-struct-builder": "^0.7.42",
     "@types/jest": "^29.5.12",
     "@types/node": "^18",
     "@typescript-eslint/eslint-plugin": "^7",
     "@typescript-eslint/parser": "^7",
+    "aws-cdk": "^2",
```
Comment (scottschreckengaust) [inline package.json:59]: Might the dependency tree take care of this?

```suggestion
```

https://www.npmjs.com/package/@aws-cdk/integ-tests-alpha?activeTab=dependencies
https://www.npmjs.com/package/@aws-cdk/integ-runner?activeTab=dependencies

Comment (scottschreckengaust) [reply package.json:59]: FYI: above is non-blocking

**Review by scottschreckengaust — COMMENTED** [review]
reviewing builds

File: .gitignore (hunk 2 above)
Comment (aws-rafams) [reply .gitignore:60]: Most of these modifications are done automatically by projen. An `AwsCdkConstructLibrary` has by default `integrationTestAutoDiscover: true` which means that as soon as an `integ.ts` file is detected, appropriate commands for deploying/destroyin/asserting/... are created under package.json as well as modifications to files such as `.gitignore` and some other files . You can try to add a file named `my-test.integ.ts`, then run build and see how the related files change.

File: package.json (hunk 3 above)
Comment (scottschreckengaust) [reply package.json]: I apologized and assumed incorrectly, looks like the build automagically adds `"aws-cdk: "^2",` it back in. Looks like https://github.com/awslabs/generative-ai-cdk-constructs/pull/663/commits/aafbd5489b7c836ab3b00b7b231c49610a70c1e4 can be reverted.

File: .projenrc.ts [hunk 4]
```diff
@@ -80,14 +81,14 @@ const project = new awscdk.AwsCdkConstructLibrary({
   npmAccess: NpmAccess.PUBLIC,
 
   publishToPypi: {
-    distName: PUBLICATION_NAMESPACE+'.'+PROJECT_NAME,
-    module: (PUBLICATION_NAMESPACE.replace(/-/g, '_'))+'.'+(PROJECT_NAME.replace(/-/g, '_')), // PEP 8, convert hypens
+    distName: PUBLICATION_NAMESPACE + '.' + PROJECT_NAME,
+    module: (PUBLICATION_NAMESPACE.replace(/-/g, '_')) + '.' + (PROJECT_NAME.replace(/-/g, '_')), // PEP 8, convert hypens
```
Comment (scottschreckengaust) [inline .projenrc.ts:85]: Adding the ESlint configuration around line 215 will fix future format spacing like here

```
project.eslint?.addRules({ 'space-infix-ops': ['error', { int32Hint: false }] });
```

File: package.json (hunk 3 above)
Comment (krokoko) [reply package.json:59]: @scottschreckengaust just reverted the commit, build is green

**Review by krokoko — DISMISSED** [review]

File: .projenrc.ts [hunk 5]
```diff
@@ -113,6 +113,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
       },
     },
   },
+  integrationTestAutoDiscover: true,
```
Comment (aws-rafams) [inline .projenrc.ts:116]: this is enabled by default, just to explicitly point it out

**Review by krokoko — APPROVED** [review]

**Review by dineshSajwan — APPROVED** [review]
Thank you @aws-rafams . This looks great.

**Pull request merged**